flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/GenericHiveMetastoreCatalog.java

```diff
@@ -229,31 +231,91 @@ public void alterDatabase(String name, CatalogDatabase newDatabase, boolean igno
 	@Override
 	public void dropTable(ObjectPath tablePath, boolean ignoreIfNotExists)
 			throws TableNotExistException, CatalogException {
-		throw new UnsupportedOperationException();
+		try {
+			client.dropTable(tablePath.getDatabaseName(), tablePath.getObjectName(), true, ignoreIfNotExists);
```

Comment:
Can we document the meaning of "true" and add a TODO for future work?

Reply:
sure